hack/deploy_fast.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+check_ocp() {
+    local KUBECONFIG_PATH=$1
+
+    export KUBECONFIG=$KUBECONFIG_PATH
+    
+    if oc get nodes; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+deploy_cluster_and_dpu_operator() {
+   bash hack/ipu_host_deploy.sh
+   bash hack/ipu_deploy.sh 
+   bash hack/deploy_traffic_flow_tests.sh
+}
+
+deploy_local_dpu_operator() {
+   bash hack/ipu_host_deploy_post.sh
+   bash hack/ipu_deploy_post.sh
+   bash hack/deploy_traffic_flow_tests.sh
+}
+
+check_ocp "/root/kubeconfig.ocpcluster"
+result_ocp=$?
+
+check_ocp "/root/kubeconfig.microshift"
+result_microshift=$?
+
+# Check if both OCP and MicroShift checks succeeded
+if [ $result_ocp -eq 0 ] && [ $result_microshift -eq 0 ]; then
+    echo "Both OCP and MicroShift cluster checks succeeded!"
+    deploy_local_dpu_operator
```

Comment:
where is the undeploy part?

Reply:
We do not need it... here we are deploying dpu operator using 'post' in CDA. When deploying the dpu operator, CDA already takes care of the idempotent issue by doing a `make-undeploy` prior to deploying. See [here](https://github.com/bn222/cluster-deployment-automation/blob/65be967c565f97cf87322f6858b5f18f3b3efb16/extraConfigDpu.py#L127). By using the post config we can take advantage of this feature without specifying it in this script.

Reply:
ok sgtm

Reply:
In the case of the IPU there are still artifacts that are not cleaned up by "make-undeploy", specifically the p4 container. We need to make this idempotent in CDA:

```
[root@238-acc ~]# podman ps
CONTAINER ID  IMAGE                                                                         COMMAND     CREATED            STATUS            PORTS       NAMES
7e2630fc7e0b  wsfd-advnetlab239.anl.eng.bos2.dc.redhat.com:5000/intel-ipu-p4-sdk:10-9-2024              About an hour ago  Up About an hour              suspicious_ishizaka
7f2e5b924393  wsfd-advnetlab239.anl.eng.bos2.dc.redhat.com:5000/intel-ipu-p4-sdk:10-9-2024              43 minutes ago     Up 43 minutes                 reverent_kapitsa
a83bc282138a  wsfd-advnetlab239.anl.eng.bos2.dc.redhat.com:5000/intel-ipu-p4-sdk:10-9-2024              25 minutes ago     Up 25 minutes                 jovial_hodgkin
392b5faefce5  wsfd-advnetlab239.anl.eng.bos2.dc.redhat.com:5000/intel-ipu-p4-sdk:10-9-2024              8 minutes ago      Up 8 minutes                  nice_banzai
```